src/components/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,71 @@
+import { ComponentProps } from 'react';
+import { getCommonProps } from '../../utils';
+import classnames from 'classnames';
+import * as Tabs from '@radix-ui/react-tabs';
+
+// Define the type for each tab, which includes its label, value, and content
+export interface Tab {
+  /**
+   * button label
+   */
+  label: string;
+  /**
+   * Button value
+   */
+  value: string;
+  /**
+   * Content to display on tab selection
+   */
+  content?: React.ReactNode;
+}
+
+/**
+ * TabsComponent renders a tab interface allowing switching between different content.
+ *
+ * @param {TabsComponentProps} props - The component props containing an array of tabs.
+ * @returns {JSX.Element} The rendered tab component.
+ */
+export interface TabsComponentProps extends ComponentProps<'div'> {
+  tabs: Tab[];
+  /**
+   * Specify the default tab
+   */
+  defaultValue?: string;
+}
```

Comment:
We may want to have an optional onClick handler that we call with the tab value for cases like updating the URL with the open tab